src/client/loop.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-console */
+
+import OpenAI from 'openai';
+import dotenv from 'dotenv';
+import debug from 'debug';
+
+import { Client } from '@modelcontextprotocol/sdk/client/index.js';
+import { StdioClientTransport } from '@modelcontextprotocol/sdk/client/stdio.js';
+
+import { packageJSON, resolveFromRoot } from '../package.js';
+
+import type { Tool } from '@modelcontextprotocol/sdk/types.js';
+
+dotenv.config();
+
+const doneTool: OpenAI.Chat.Completions.ChatCompletionTool = {
+  type: 'function',
+  function: {
+    name: 'done',
+    description: 'Call this tool to indicate that the task is complete.',
+    parameters: {},
+  },
+};
+
+async function connectToMCP(): Promise<{ client: Client, tools: Tool[] }> {
+  const transport = new StdioClientTransport({
```

Comment:
If you want to use it for testing, maybe an in-memory transport is better because it allows easier debugging.
https://github.com/modelcontextprotocol/typescript-sdk/blob/main/src/inMemory.ts